code/datums/components/chasm.dm

```diff
@@ -58,55 +62,71 @@ GLOBAL_LIST_INIT(chasm_storage, list())
 		chasm_storage += ref
 	GLOB.chasm_storage = chasm_storage
 
-/datum/component/chasm/proc/Entered(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+/datum/component/chasm/proc/entered(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
 	SIGNAL_HANDLER
+	addtimer(CALLBACK(src, PROC_REF(try_drop), arrived), 0)
 
-	START_PROCESSING(SSobj, src)
-	drop_stuff(arrived)
+/datum/component/chasm/proc/exited(datum/source, atom/movable/exited)
+	SIGNAL_HANDLER
+	UnregisterSignal(exited, list(COMSIG_MOVETYPE_FLAG_DISABLED, COMSIG_LIVING_SET_BUCKLED, COMSIG_MOVABLE_THROW_LANDED))
 
-/datum/component/chasm/process()
-	if (!drop_stuff())
-		STOP_PROCESSING(SSobj, src)
+/datum/component/chasm/proc/initialized_on(datum/source, atom/movable/movable, mapload)
+	SIGNAL_HANDLER
+	addtimer(CALLBACK(src, PROC_REF(try_drop), movable), 0) //give it the time to finish initializing at least.
```

Comment:
😭 I don't have any smart ideas for this one

Maye you could... add `COMSIG_ATOM_AFTER_SUCCESSFUL_INITIALIZED_ON` ?
That should give time for a trait applied by an item initialising on this turf to be applied to the turf if it is sent at a similar time to `COMSIG_ATOM_AFTER_SUCCESSFUL_INITIALIZE`

Reply:
I should move the `initialized_on` signal outside of the initialized proc, perhaps called in the atoms subsystem AFTER the atom has finished initializing.

Reply:
Ok sir, I've moved the signal next to COMSIG_ATOM_AFTER_SUCCESSFUL_INITIALIZE and renamed it COMSIG_ATOM_AFTER_SUCCESSFUL_INITIALIZED_ON.